vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
+    return (in - low) * out_range / in_range + low_out;
 }
 
-HAL_StatusTypeDef initMotorControllerProcanSettings()
+HAL_StatusTypeDef initMotorControllerSettings()
 {
-    mcLeftSettings.InverterCommand = 0;
-    mcLeftSettings.DriveTorqueLimit = MAX_TORQUE_DEMAND_DEFAULT;
-    mcLeftSettings.BrakingTorqueLimit = BRAKING_TORQUE_LIMIT_DEFAULT;
-    mcLeftSettings.ForwardSpeedLimit = SPEED_LIMIT_DEFAULT;
-    mcLeftSettings.ReverseSpeedLimit = SPEED_LIMIT_DEFAULT;
-    mcLeftSettings.DischargeCurrentLimit = DISCHARGE_CURRENT_LIMIT_DEFAULT;
-    mcLeftSettings.ChargeCurrentLimit = CHARGE_CURRENT_LIMIT_DEFAULT;
-    mcLeftSettings.HighVoltageLimit = HIGH_VOLTAGE_LIMIT_DEFAULT;
-    mcLeftSettings.LowVoltageLimit = LOW_VOLTAGE_LIMIT_DEFAULT;
-    mcLeftSettings.MaxTorqueDemand = MAX_TORQUE_DEMAND_DEFAULT;
-
-    mcRightSettings.InverterCommand = 0;
-    mcRightSettings.DriveTorqueLimit = MAX_TORQUE_DEMAND_DEFAULT;
-    mcRightSettings.BrakingTorqueLimit = BRAKING_TORQUE_LIMIT_DEFAULT;
-    mcRightSettings.ForwardSpeedLimit = SPEED_LIMIT_DEFAULT;
-    mcRightSettings.ReverseSpeedLimit = SPEED_LIMIT_DEFAULT;
-    mcRightSettings.DischargeCurrentLimit = DISCHARGE_CURRENT_LIMIT_DEFAULT;
-    mcRightSettings.ChargeCurrentLimit = CHARGE_CURRENT_LIMIT_DEFAULT;
-    mcRightSettings.HighVoltageLimit = HIGH_VOLTAGE_LIMIT_DEFAULT;
-    mcRightSettings.LowVoltageLimit = LOW_VOLTAGE_LIMIT_DEFAULT;
-    mcRightSettings.MaxTorqueDemand = MAX_TORQUE_DEMAND_DEFAULT;
-
+    mcSettings.InverterMode = 0;
+    mcSettings.DriveTorqueLimit = MAX_TORQUE_DEMAND_DEFAULT_NM; 
+    mcSettings.MaxTorqueDemand = MAX_TORQUE_DEMAND_DEFAULT_NM;
+    mcSettings.DirectionCommand = INVERTER_DIRECTION_FORWARD;
+
+    // TODO - legacy settings - do we still need?
+    mcSettings.ForwardSpeedLimit = SPEED_LIMIT_DEFAULT;
+    mcSettings.DischargeCurrentLimit = DISCHARGE_CURRENT_LIMIT_DEFAULT;
+    mcSettings.ChargeCurrentLimit = CHARGE_CURRENT_LIMIT_DEFAULT;
     return HAL_OK;
 }
 
-HAL_StatusTypeDef setMotorControllerProcanSettings(MotorControllerProcanSettings settings)
+HAL_StatusTypeDef setMotorControllerSettings(MotorControllerSettings settings)
 {
-    memcpy(&mcLeftSettings, &settings, sizeof(settings));
-    memcpy(&mcRightSettings, &settings, sizeof(settings));
-
+    memcpy(&mcSettings, &settings, sizeof(settings));
     return HAL_OK;
 }
 
 HAL_StatusTypeDef setForwardSpeedLimit(float limit)
 {
-    mcLeftSettings.ForwardSpeedLimit = limit;
-    mcRightSettings.ForwardSpeedLimit = limit;
-
+    mcSettings.ForwardSpeedLimit = limit;
     return HAL_OK;
 }
 
 HAL_StatusTypeDef setTorqueLimit(float limit)
 {
-    mcRightSettings.DriveTorqueLimit = limit;
-    mcLeftSettings.DriveTorqueLimit = limit;
-
+    mcSettings.DriveTorqueLimit = limit;
     return HAL_OK;
 }
 
 HAL_StatusTypeDef setDischargeCurrentLimit(float limit)
 {
-    mcRightSettings.DischargeCurrentLimit = limit;
-    mcLeftSettings.DischargeCurrentLimit = limit;
+    mcSettings.DischargeCurrentLimit = limit;
```

Comment:
Are all these parameters that we can set with the new MC? Don't need to keep them all if there isnt a setting to control these directly

Reply:
With that said, for formula electric we will need to add a 80kW limit. Can you add that functionality via a macro